libs/blocks/modal/modal.js

```diff
@@ -202,9 +208,17 @@ export async function getModal(details, custom) {
 
   const iframe = dialog.querySelector('iframe');
   if (iframe) {
+    if (details?.title) iframe.setAttribute('title', details.title);
+
     if (dialog.classList.contains('commerce-frame') || dialog.classList.contains('dynamic-height')) {
       const { default: enableCommerceFrameFeatures } = await import('./modal.merch.js');
       await enableCommerceFrameFeatures({ dialog, iframe });
+
+      if (!details?.title) {
+        const commerceDetails = findDetails(window.location.hash, null);
+        const commerceFrameTitle = commerceDetails?.title ? commerceDetails.title : null;
```

Comment:
Nit: `const commerceFrameTitle = commerceDetails?.title || null;` would be shorter and also account for `''` as a title.